internal/pkg/client/oci/nativesif.go

```diff
@@ -13,23 +13,21 @@ import (
 	"github.com/sylabs/singularity/v4/internal/pkg/build"
 	"github.com/sylabs/singularity/v4/internal/pkg/cache"
 	"github.com/sylabs/singularity/v4/internal/pkg/ociimage"
+	"github.com/sylabs/singularity/v4/internal/pkg/ocitransport"
 	buildtypes "github.com/sylabs/singularity/v4/pkg/build/types"
 	"github.com/sylabs/singularity/v4/pkg/sylog"
 )
 
 // pullNativeSIF will build a SIF image into the cache if directTo="", or a specific file if directTo is set.
 func pullNativeSIF(ctx context.Context, imgCache *cache.Handle, directTo, pullFrom string, opts PullOptions) (imagePath string, err error) {
-	sys, err := sysCtx(opts)
-	if err != nil {
-		return "", err
-	}
+	to := transportOptions(opts)
```

Comment:
Another nit: I personally don't love `to` as the name for a variable. I realize it's not a keyword in Go, but it is very keyword-*ish*, if that makes sense. It's really hard to read it as anything other than the English preposition "to." How about something like `tOpts`?

(Comment applies in numerous other places in the PR, but I didn't want to pollute the review with repetitions of this, so just do a search if you decide to implement.)

Reply:
Yeah, probably fair. Especially when we're in the realm of moving imaged from somewhere, to somewhere.